sql/core.go

```diff
@@ -341,6 +341,18 @@ type CheckAlterableTable interface {
 	DropCheck(ctx *Context, chName string) error
 }
 
+// StatementBounder allows the boundaries of an INSERT, REPLACE, UPDATE, or DELETE statement to be passed along.
+// This is modeled after the SAVEPOINT statements, but are local and specific to the aforementioned data manipulation
+// statement boundaries.
+type StatementBounder interface {
+	// Savepoint is called before the first operation of a statement.
+	Savepoint(ctx *Context)
```

Comment:
Comments here need work. What's the contract of these methods? What do integrators need to do when these are called?